R/with.R

```diff
@@ -51,7 +51,7 @@ modify_defaults <- function(defaults, ...) {
 
   res[] <- lapply(res, function(x) {
     prev_class <- class(x)
-    if (is.function(x) && !is_linter(x)) {
+    if (is.function(x) && !is_linter_factory(x) && !is_linter(x)) {
```

Comment:
Good catch!

Can we extract these conditionals into a boolean function with an expressive name and use that instead? 
I think that will make this code more readable. And, as a side effect, it might produce a reusable, internally useful helper function. WDYT?

E.g.

## current

```r
if (looks_like_a_duck(x) && swims_like_a_duck(x) && quacks_like_a_duck(x)) {
  ...
}
```

## proposed

```r
is_duck <- function(x) {
  looks_like_a_duck(x) &&
    swims_like_a_duck(x) &&
    quacks_like_a_duck(x)
}

if (is_duck(x)) {
  ...
}
```

Reply:
Brilliant!

No code is always more readable than any code 😉